### PR TITLE
add location list to restaurants

### PR DESCRIPTION
**Pull request by hadarbu**

### CODE REVIEW

File: openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java [hunk 1]
```diff
@@ -43,6 +44,8 @@ public Restaurant(String id, String alias, String affiliateId, Map<String, Strin
         this.reservations = reservations;
         this.features = features;
         this.currentLocationId = currentLocationId;
+        this.defaultLocationIdAtTimeOfMigration = defaultLocationIdAtTimeOfMigration;
+        this.locationName = locationName;
```
Comment (alexgask) [inline openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java:48]: why is that required?

File: openrest4j-api/src/main/java/com/openrest/v1_1/Location.java [hunk 2]
```diff
@@ -59,23 +85,80 @@ public boolean equals(Object o) {
...
+    public List<DispatchInfo> deliveryInfos = new LinkedList<>();
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public DeliveriesInfo deliveriesInfo;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public OrdersInfo orders;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public ReservationsInfo reservations;
```
Comment (alexgask) [inline openrest4j-api/src/main/java/com/openrest/v1_1/Location.java:161]: I am not sure it is used or what is it for

File: openrest4j-api/src/main/java/com/openrest/v1_1/Location.java [hunk 3]
```diff
@@ -59,23 +85,80 @@ public boolean equals(Object o) {
...
+
+    /** Information regarding the different delivery destinations. */
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    public List<DispatchInfo> deliveryInfos = new LinkedList<>();
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public DeliveriesInfo deliveriesInfo;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public OrdersInfo orders;
```
Comment (alexgask) [inline openrest4j-api/src/main/java/com/openrest/v1_1/Location.java:158]: I am not sure it is used or what is it for

File: openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java (hunk 1 above)
Comment (hadarbu) [reply openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java:48]: If we'll need to return the title(business name)  and the locationName at the object.

File: openrest4j-api/src/main/java/com/openrest/v1_1/Location.java (hunk 2 above)
Comment (hadarbu) [reply openrest4j-api/src/main/java/com/openrest/v1_1/Location.java:161]: This is the settings of the reservation, it will change to be per location in the client.

File: openrest4j-api/src/main/java/com/openrest/v1_1/Location.java (hunk 3 above)
Comment (hadarbu) [reply openrest4j-api/src/main/java/com/openrest/v1_1/Location.java:158]: This is the settings of the orders, it will change to be per location in the client also.

File: openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java (hunk 1 above)
Comment (alexgask) [reply openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java:48]: But why shouldn't it be added on the fly?
Why should we store it (what is the difference from what we said about currentLocationId)?

Comment (hadarbu) [reply openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java:48]: right, i'll change it.

Comment (alexgask) [reply openrest4j-api/src/main/java/com/openrest/v1_1/Restaurant.java:48]: Actually I think I get it...
